platforms/ide/tooling-api/src/main/java/org/gradle/tooling/events/test/source/UriSource.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.tooling.events.test.source;
+
+
+import org.gradle.api.Incubating;
+
+import java.net.URI;
+
+/**
+ * A test source that can be identified by a URI.
+ *
+ * @since 9.4.0
+ */
+@Incubating
+public interface UriSource extends TestSource {
```

Comment:
I'm not sure we need this one right now either since we don't have a way of selecting URI-based tests, so the only way we'd see this is if a test engine created it directly